### ISSUE TITLE
CNTK and semantic segmentation

### ISSUE THREAD

**Issue by julienlopez** [Microsoft/CNTK#1613]
Hi, I'm trying to do some very basic semantic segmentation with a fully conv net, but I'm having issue with the output:
First of all, I'm not using deconvolutions/unpooling yet, I'm restricting myself to just stacking layers of convolutions/pooling so that my output feature map is much smaller than my input (26x18 in my case).

But the problem I have is with how to encode the classes in the output.

My Naive approach would be to use some kind of pixel-wise softmax along the depth of the output (where the depth would be equal to the number of classes then).
Or should forget the softmax and just use sigmoids? in this case, does rsme work with 3d outputs?

In any case, anyone has any idea how this is done is already existing solutions? I read through many papers on the subject but the actual encoding of the output and the cost function is not really detailed as far as I could find.

Thank you.

**Comment by vijtad** [comment]
Is there any example on Semantic segmentation ?
I am very new to CNTK.
I can see examples of Tensorflow using FCNs of image segmentation using PSACAL VOC.

**Comment by Sangrail** [comment]
I would really like to see an example of Semantic Segmentation - especially given the other toolkits support this.

**Comment by spandantiwari** [comment]
We will offer a semantic segmentation example soon. Please stay tuned.

**Comment by julienlopez** [comment]
Awesome, can't wait to see it :)

**Comment by Mut1nyJD** [comment]
Hello 
I am just wondering what happened to this? 
Originally I saw this one the road map for 2.3 but we are now at 2.5 and no example has popped up in the mean time.
I guess the main problem I am seeing is how do you define a per-pixel loss function and having a label-matrix instead of just a label vector?  I saw in 2.5 the hierarchical_softmax_layer has appeared could that help?
